src/Orleans/Async/Cancellation/CancellationSourcesExtension.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Threading.Tasks;
+using Orleans.Runtime;
+
+namespace Orleans.Async
+{
+    /// <summary>
+    /// Contains list of cancellation token source corresponding to the tokens
+    /// passed to the related grain activation.
+    /// </summary>
+    internal class CancellationSourcesExtension : ICancellationSourcesExtension
+    {
+        private readonly Lazy<TraceLogger> _logger = new Lazy<TraceLogger>(() =>
+            TraceLogger.GetLogger("CancellationSourcesExtension", TraceLogger.LoggerType.Application));
+        private static readonly Interner<Guid, GrainCancellationTokenSource> _cancellationTokenSources;
```

Comment:
The `GrainCancellationToken` is only holder of the `CancellationToken`, which doesn't expose cancel method either.